azurerm/resource_arm_data_lake_store.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDateLakeStoreRead(d, meta)
+}
+
+func resourceArmDateLakeStoreUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	newTags := d.Get("tags").(map[string]interface{})
+	newTier := d.Get("tier").(string)
+
+	props := account.UpdateDataLakeStoreAccountParameters{
+		Tags: expandTags(newTags),
+		UpdateDataLakeStoreAccountProperties: &account.UpdateDataLakeStoreAccountProperties{
+			NewTier: account.TierType(newTier),
+		},
+	}
+
+	future, err := client.Update(ctx, resourceGroup, name, props)
+	if err != nil {
+		return fmt.Errorf("Error issuing update request for Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for the update of Data Lake Store %q (Resource Group %q) to commplete: %+v", name, resourceGroup, err)
+	}
+
+	return resourceArmDateLakeStoreRead(d, meta)
+}
+
+func resourceArmDateLakeStoreRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] DataLakeStoreAccount '%s' was not found (resource group '%s')", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if tier := resp.DataLakeStoreAccountProperties; tier != nil {
+		d.Set("tier", string(tier.CurrentTier))
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDateLakeStoreDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func validateArmDataLakeName(v interface{}, k string) (ws []string, es []error) {
```

Comment:
Very minor: as it doesn't look like this is used anywhere else you could just use the built in validation helpers:
```go
                ValidateFunc: validation.StringMatch(
					regexp.MustCompile(`\A([a-z0-9]{3,24})\z`),
					"Name can only consist of lowercase letters and numbers, and must be between 3 and 24 characters long",
				),
```
But I'll merge as is

Reply:
makes sense, no need for a func that doesnt get used elsewhere